### PR TITLE
Update espresso-jshell demo.

### PR DESCRIPTION
**Pull request by mukel**
This PR is an update for building the demo with Java 17 and documentation improvements.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
         run: |
           cd espresso-jshell
           ./build-espresso-jshell.sh
-          echo "Math.PI" | ./espresso-jshell
+          echo "Math.PI" | ./espresso-jshell -Dorg.graalvm.home=$GRAALVM_HOME
```
Comment (fniephaus) [inline .github/workflows/main.yml:27]: Why does `$GRAALVM_HOME` need to be passed in? Is there no way to detect this from inside the `espresso-jshell`?

File: espresso-jshell/README.md [hunk 2]
```diff
@@ -1,51 +1,56 @@
...
-You will build a native executable of `jshell`, that executes dynamically generated bytecode on Espresso.
-This hybrid mode achieves instant startup, beating the vanilla `jshell` in both: time to the first interaction and time to evaluate a simple expression.
+This demo showcases the integration between [GraalVM Native Image](https://www.graalvm.org/reference-manual/native-image/) and [Java-on-Truffle (Espresso)](https://www.graalvm.org/reference-manual/java-on-truffle/).  
+It builds a native image of `jshell`, that executes the dynamically generated bytecodes on Espresso. This hybrid mode achieves instant startup, beating the vanilla `jshell` in both: time to the first interaction and time to evaluate a simple expression.
 
-JShell is a Java read-eval-print loop tool first introduced in the JDK 9, but this demo also shows how to run `jshell` on Java 8.
+JShell is a Java read-eval-print loop tool first introduced in Java 9, this demo also allows to run `jshell` on Java 8.
```
Comment (fniephaus) [inline espresso-jshell/README.md:6]:
```suggestion
JShell is a Java read-eval-print loop tool first introduced in Java 9, this demo also allows running `jshell` on Java 8.
```

Comment (mukel) [reply espresso-jshell/README.md]: I manually fixes this nits in the latest version.

File: .github/workflows/main.yml (hunk 1 above)
Comment (mukel) [reply .github/workflows/main.yml]: Fixed! Now it tries to set/initialize `org.graalvm.home` and `java.home` from the `GRAALVM_HOME` env. variable.
Running just `./espresso-jshell` may give the false illusion that the executable is fully standalone.
I added a small section to the README explaining this.

**Review by fniephaus — APPROVED** [review]

File: espresso-jshell/README.md [hunk 3]
```diff
@@ -1,51 +1,58 @@
 # GraalVM Demos: Native jshell and Espresso
 
-This demo showcases the integration between [GraalVM Native Image](https://www.graalvm.org/reference-manual/native-image/) and [Java-on-Truffle (Espresso)](https://www.graalvm.org/reference-manual/java-on-truffle/).
-You will build a native executable of `jshell`, that executes dynamically generated bytecode on Espresso.
-This hybrid mode achieves instant startup, beating the vanilla `jshell` in both: time to the first interaction and time to evaluate a simple expression.
+This demo showcases the integration between [GraalVM Native Image](https://www.graalvm.org/reference-manual/native-image/) and [Java-on-Truffle (Espresso)](https://www.graalvm.org/reference-manual/java-on-truffle/).  
+It builds a native image of `jshell`, that executes the dynamically generated bytecodes on Espresso. This hybrid mode achieves instant startup, beating the vanilla `jshell` in both: time to the first interaction and time to evaluate a simple expression.
```
Comment (olyagpl) [inline espresso-jshell/README.md:4]: bytecodes > bytecode

File: espresso-jshell/README.md [hunk 4]
```diff
@@ -1,51 +1,58 @@
...
+
+JShell is a Java read-eval-print loop tool first introduced in Java 9, this demo also allows running `jshell` on Java 8.
```
Comment (olyagpl) [inline espresso-jshell/README.md:6]: JShell is a Java read-eval-print loop tool first introduced in Java 9, this demo also allows running `jshell` on Java 8.  

JShell is a Java "read-eval-print loop tool" first introduced in Java 9, and this demo also allows running `jshell` on Java 8.

**Pull request merged**